### DISCUSSION TITLE
"code" argument must be single file (expect for "package")?

### DISCUSSION THREAD

**Discussion by cpierceworld** [explosion/spaCy#10694] [Help: Other Questions] [answered]
I'm trying to include custom components that I have defined in multiple files and include them with the "code" command line argument.

Problem is only "package" accepts a comma separated list of files.   The rest only accept 1 file ("train", "pretrain", "evaluate" and "assemble").

Does anyone know if there is a particular reason for that?  

I really don't want to put all my components into one monolithic file.
(Or better for me would be to specify the root folder of a python package and have it use/copy the whole package)

**Comment by thomashacker** [comment, marked as answer]
Hello,
For these commands (`train`, `pretrain`, `evaluate`, and `assemble`) you'll only need one `.py` file which just imports the custom code. For the `package` command however, you need to include every custom code file to the CLI. 

The Healthsea spaCy project is a good example for that:

- [CLI command for train](https://github.com/explosion/healthsea/blob/main/project/project.yml#L135)
- [Custom code file for train](https://github.com/explosion/healthsea/blob/main/project/scripts/clausecat/custom_code.py)
- [CLI command for packaged](https://github.com/explosion/healthsea/blob/main/project/project.yml#L161)

**Reply by vitorbaptista** [reply]
Is this something you'd consider changing (or accepting a PR that does so)?

Currently, I need to create a single file that imports every required file just for the `train`, `pretrain`, etc., while in `package` I can just import everything directly. Is there a reason for this inconsistency?

**Reply by thomashacker** [reply]
Yes, this is something we're considering changing in the future, and we're happy to accept a PR for that.
